Blog: Do your ads respect privacy (#202)
readthedocs/ethicalads.io: content/posts/2021-do-the-ads-on-your-site-respect-privacy.md

Title: Do the Ads on your Site Respect Privacy

> Date: October 10, 2021

Author: ```suggestion
Date: October 12, 2021
```

Title: Do the Ads on your Site Respect Privacy
Date: October 10, 2021

> description: There's a lot of ad networks out there to choose from but none of them are as focused on privacy as EthicalAds.

Reviewer: ```suggestion
description: There's a lot of ad networks out there, but they don't make the same privacy guarantees as EthicalAds.
```

Original is a bit passive.

tags: cookies, privacy, publishers
authors: David Fischer
image: /images/posts/2021-privacy-respect.jpg
image_credit: <span>Photo by <a href="https://unsplash.com/@matthewhenry?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">Matthew Henry</a> on <a href="https://unsplash.com/photos/fPxOowbR6ls?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">Unsplash</a></span>

> Ads fund a lot of the websites across the internet just as advertising
> was used in the business models of TV and radio in the past.

Reviewer: ```suggestion
Ads fund a lot of the websites across the internet, 
comparable to the business models of TV and radio in the past.
```

Unlike in previous eras, however, web technology combined with improvements in data storage
have allowed an unprecedented ability to target and personalize ads to individuals.
This kind of tracking has enabled the rise of
[surveillance advertising]({filename}../pages/learning-hub/surveillance-advertising.md).

> Are the ads you're running on your site spying on you?

Reviewer: This feels a bit abrupt. I feel like we need a bit more of a leadup to this, or some more context. 

```suggestion

As a publisher, you take pride in the work you put on the internet.
However, with traditional approaches to advertising, your users could be sharing more with ad networks than you think. 
Are the ads you're running on your site spying on you?
```

In this post, we'll show you exactly what to watch out for and how to choose alternatives

> to invasive surveillance for your site.

Reviewer: This feels like a strong words, and I might feel attacked as a publisher.

Author: I'm going to remove this paragraph and have a checklist of things to watch out for with an ad network.

## Advertiser provided cookies, code, or resources

Probably the biggest things to watch out for when selecting an ad network
are advertiser provided resources, code, or cookies.
If an advertiser hosts the images or other resources for an advertisement,
they can cookie and track the users on your site across the web
(and then [show ads to your users more cheaply elsewhere]({filename}2021-invasive-ad-targeting-bad-journalism-premium-publishers.md)).

<div class="postimage">

>   <img class="w-100" src="{static}../images/posts/2021-third-party-resources.png" alt="An ad network loading third-party resources">

Author: This image is a bit of a shot across the bow.

  <p>An ad network loading third-party resources.</p>
</div>

Even worse than loading a tracking pixel is to run JavaScript code
provided by advertisers.
In addition to being a security nightmare,
this allows for even more exfiltration of data from your site
including some cookie values or logged-in user data displayed on your site.
While common a decade ago, thankfully allowing advertisers to run code on publisher sites

> has become rarer today.

Reviewer: Does Google allow this? Should we name specific people doing this?

Author: They used to. I don't think they do anymore.

With EthicalAds, we vet all our advertisers
but we still never let them run arbitrary code or cookie users on your site.
We don't save a cookie from our domains either.
Advertising can work without knowing any personal details or past sites you've visited.
With us, you don't need a [cookie warning banner]({filename}2021-can-you-remove-cookie-banners.md)
and the ads on your site will already comply with privacy regulation like the GDPR.

## Fingerprinting, supercookies, and other nefarious tracking tech

Some common browsers like Firefox and Safari already block many third-party cookies.
This is a great step toward a more private web.
However, to "work around" a lack of cookies, some ad networks use fingerprinting
in an attempt to uniquely identify visitors for ad targeting.

Fingerprinting and other tracking tech can take a few forms but some of the most common are:

> * **Browser Fingerprinting:** [Browser fingerprinting](https://ssd.eff.org/en/module/what-fingerprinting),

Reviewer: Love this checklist 💯

  and other data about them like their IP address.
  These browser attributes can be anything from how the browser identifies itself
  to settings you've changed to protect your privacy.
  Customizations can make your browser appear unique rather than look like everybody else.
* **Supercookies:** So called ["supercookies"](https://blog.mozilla.org/security/2021/01/26/supercookie-protections/)
  use trackers hidden in other parts of browser like local storage, ETags, or HSTS headers.
  These are not cleared by clearing cookies
  and can potentially identify users uniquely or be used in combination with fingerprinting.
* **CNAME Cloaking:** [CNAME Cloaking](https://webkit.org/blog/11338/cname-cloaking-and-bounce-tracking-defense/)
  involves a publisher setting up a subdomain for an ad tech tracker.
  This can make a third-party cookie look like a first-party cookie and gets around most tracking protection.

Some of these tracking methods are easy to spot.
If you see data from your ad network in local storage, that's a red flag.
However, for some forms of fingerprinting, it happens only on the ad network's servers

> and you have no way of knowing.

Reviewer: This is.. not super satisfying. I'd love to see something more actionable here, like a set of questions you can ask your ad provider.

EthicalAds doesn't use any fingerprinting technology.
You don't need to trust us. All our code
&mdash; both the [client](https://github.com/readthedocs/ethical-ad-client) that runs on your site
and the [server](https://github.com/readthedocs/ethical-ad-server) serving the ads &mdash; is open source.
We don't track users to target them with ads because we're confident in our contextual targeting.

> ## Contextual targeting works

Reviewer: We need a longer piece of content on this. We've linked some great blog posts and learning hub content in this post, but it's a shame we don't have anything to link here other than our vision. Something that shows results, perhaps a case study is the best option, or re-framing one of our testimonials a bit more towards this.
